### PR TITLE
HPCC-13525 Added whitelist file patterns for testing directory

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner Michael.Gardner@lexisnexis.com

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@ghalliday @AttilaVamos Please review when you have time.  This commit will make it so we no longer add extraneous files to the package when a user attempts to run the regression suite from source.  If there are any patterns that the output of the regression tests would fit which get covered by my patterns, please give me a heads up.

File: testing/CMakeLists.txt [hunk 1]
```diff
@@ -17,6 +17,9 @@ HPCC_ADD_SUBDIRECTORY (unittests)
...
          USE_SOURCE_PERMISSIONS
-         PATTERN regress/ecl EXCLUDE
-         PATTERN regress/environment.xml.in EXCLUDE
-         PATTERN regress/CMakeLists.txt EXCLUDE )
+         FILES_MATCHING
+         PATTERN "regress/ecl-test"
+         PATTERN "regress/ecl-test.json"
+         PATTERN "regress/environment.xml"
+         PATTERN "regress/README.rst"
+         PATTERN regress/ecl EXCLUDE )
```
Comment (AttilaVamos) [inline testing/CMakeLists.txt]: @Michael-Gardner It is possible I missed something, but why 'regress/ecl' excluded?

Comment (Michael-Gardner) [inline testing/CMakeLists.txt]: I don't know why it was originally excluded.  I merely kept the directory structure the same with the new rules.

Comment (AttilaVamos) [inline testing/CMakeLists.txt]: @richardkchapman @ghalliday Do you know the answer?

Comment (richardkchapman) [inline testing/CMakeLists.txt]: I don't. If we are including the suite engine then it makes sense to also include the ecl and key files. Unless @ghalliday  knows differently.

**Comment by ghalliday** [comment]
I can't think of any reason why not - they're not very large, and provide something that can be used to test the system.

**Comment by Michael-Gardner** [comment]
I'll go ahead and add them to the white list then.  I'll get a new commit up with the changes asap.

**Comment by Michael-Gardner** [comment]
@ghalliday added ecl and it's subdirectories, and added a whitelist of file extensions.

File: testing/regress/CMakeLists.txt [hunk 2]
```diff
@@ -14,6 +14,21 @@
...
+
 configure_file("environment.xml.in" "environment.xml")
 
 Install ( FILES ${CMAKE_CURRENT_BINARY_DIR}/environment.xml DESTINATION "./testing/regress" COMPONENT Runtime )
+
+Install ( DIRECTORY download hpcc ecl DESTINATION "./testing/regress" COMPONENT Runtime
+          USE_SOURCE_PERMISSIONS
+          FILES_MATCHING
+          PATTERN "hpcc/__init__.py"
+          PATTERN "download/0drvb10.txt"
```
Comment (ghalliday) [inline testing/regress/CMakeLists.txt]: trivial: Could probably use download/*.txt

**Comment by ghalliday** [comment]
@richardkchapman looks good to me.  Please double check.

**Pull request merged**